### ISSUE TITLE
handling of long composed enumerators

### ISSUE THREAD

**Issue by TLiebich** [bSI-InfraRoom/IFC-Specification#207]
when looking into the proposed additions from INFRA/RAIL there is a mix of concatination and "_" separation of long terms,

e.g. in IfcDiscreteAccessoryTypeEnum

POINTMACHINEMOUNTINGDEVICE
POINT_MACHINE_LOCKING_DEVICE
RAIL_MECHANICAL_EQUIPMENT

the prefered way in IFC had been without "_", although it was never 100% obeyed and there are several existing cases with "_". But adding similar concepts, as above, sometimes with and sometime without an "_" to the same enumeration is strange. The usage style should at least be consistent within one enumeration.

**Comment by TLiebich** [comment]
similar issue in IfcDistributionSystemEnum with 

	,CATENARY_SYSTEM
	,OVERHEAD_CONTACTLINE_SYSTEM
	,RETURN_CIRCUIT
	,FIXEDTRANSMISSIONNETWORK
	,OPERATIONALTELEPHONYSYSTEM
	,MOBILENETWORK
	,MONITORINGSYSTEM

**Comment by TLiebich** [comment]
similar in IfcDoorTypeEnum

	,TRAPDOOR
	,BOOM_BARRIER
	,TURNSTILE

**Comment by TLiebich** [comment]
same in IfcEarthworksCutTypeEnum

	(TRENCH
	,DREDGING
	,EXCAVATION
	,OVEREXCAVATION
	,TOPSOILREMOVAL
	,STEPEXCAVATION
	,PAVEMENTMILLING
	,CUT
	,BASE_EXCAVATION

**Comment by TLiebich** [comment]
same in IfcProtectiveDeviceTypeEnum

	(CIRCUITBREAKER
	,EARTHLEAKAGECIRCUITBREAKER
	,EARTHINGSWITCH
	,FUSEDISCONNECTOR
	,RESIDUALCURRENTCIRCUITBREAKER
	,RESIDUALCURRENTSWITCH
	,VARISTOR
	,ANTI_ARCING_DEVICE
	,SPARKGAP
	,VOLTAGELIMITER

**Comment by TLiebich** [comment]
same for IfcSlabTypeEnum

	,BASESLAB
	,APPROACH_SLAB
	,PAVING
	,WEARING
	,SIDEWALK
	,TRACKSLAB

**Comment by TLiebich** [comment]
same for IfcSwitchingDeviceTypeEnum

	,MOMENTARYSWITCH
	,SELECTORSWITCH
	,STARTER
	,SWITCHDISCONNECTOR
	,TOGGLESWITCH
	,RELAY
	,START_AND_STOP_EQUIPMENT

**Comment by TLiebich** [comment]
same for IfcTrackElementTypeEnum

	(TRACKENDOFALIGNMENT
	,BLOCKINGDEVICE
	,VEHICLESTOP
	,SLEEPER
	,HALF_SET_OF_BLADES
	,SPEEDREGULATOR

**Comment by SergejMuhic** [comment]
This was sort of left to the author if the enumeration name in question is judged to be readable without then no underscores were used. In the opposite case, an underscore was used and especially something like `BASE_EXCAVATION` even "softly" required.

**Comment by TLiebich** [comment]
Well, I can follow this reasoning if it is applied consistently. But what about this:

POINTMACHINEMOUNTINGDEVICE
POINT_MACHINE_LOCKING_DEVICE

**Comment by TLiebich** [comment]
or when BASE_EXCAVATION is required, why not having the same requirement for OVEREXCAVATION within the same enumeration?

**Comment by SergejMuhic** [comment]
I completely agree with you. But, most of these were already in RC1, for the others I was told that the naming was agreed in the harmonization meeting (or that is how I understood it, maybe not all were agreed).
